### ISSUE TITLE
API: Timestamp/Timedelta resolution in __repr__

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#49059]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
NA
```

### Issue Description

Discussed on today's call, in 2.0 Timestamp and Timedelta will support non-nanosecond resolutions, and this should probably be reflected in the __repr__, something like `Timestamp("2016-01-01 10:24:11.012", reso="ms")`.

ATM it is possible to do `eval(repr(ts))` on a Timestamp.  If we want to maintain that, we'll need to add a reso keyword (or another name, we discussed "unit" but that conflicts with the existing unit keyword) to the constructors.

If we do add a reso keyword, it may make sense to add it to to_datetime, date_range, to_timedelta, timedelta_range.  That would avoid the need to do reso-inference in those functions.

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

**Comment by jreback** [comment]
adding reso to the constructors and repr lgtm

**Comment by jbrockmendel** [comment]
non-nano has been out for a while and no one has complained about this not being in the repr. Closing.
